main.tf

```diff
@@ -28,19 +32,15 @@ resource "aws_subnet" "public" {
     create_before_destroy = true
   }
 
-  tags {
-    Name = "${var.name}.${element(var.azs, count.index)}"
-  }
+  tags = "${merge(var.tags, map("Name", format("%s-subnet-public-%s", var.name, element(var.azs, count.index))))}"
 }
 
 # Routes
 resource "aws_route_table" "public" {
   vpc_id = "${var.vpc_id}"
   count  = "${length(var.cidrs)}"
 
-  tags {
-    Name = "${var.name}.${element(var.azs, count.index)}"
-  }
+  tags = "${merge(var.tags, map("Name", format("%s-rt-public-%s", var.name, element(var.azs, count.index))))}"
```

Comment:
A problem I encounter with this approach it's resources won't be named by their type so for example if I pick as the variable `name = "myproject"` then my resources won't have a descriptive name and if I choose as an example for this case `name = "myproject.subnet.public"` expecting to have a descriptive name for the resources generated by this module I'll end having a route table with a name that doesn't correspond to the resource e.g. `"myproject.subnet.public.us-east-1a"`